aten/src/ATen/core/LazyTensor.cpp

```diff
@@ -0,0 +1,142 @@
+#include <ATen/core/LazyTensor.h>
+#include <ATen/core/function_schema.h>
+#include <torch/csrc/autograd/function.h>
+#include <torch/csrc/autograd/variable.h>
+#include <torch/csrc/jit/autodiff.h>
+#include <torch/csrc/jit/graph_executor.h>
+#include <torch/csrc/jit/interpreter.h>
+#include <torch/csrc/jit/operator.h>
+#include <torch/csrc/jit/script/function_schema_parser.h>
+
+namespace at {
+
+// Embed the LazyTensorImpl into the passed in graph and
+// update map of Value*'s in the graph to their materialized
+// IValues.
+//
+// Returns a Value* in the graph.
+//
+// TODO: Make this non-recursive
+torch::jit::Value* getValue(
+    const LazyTensorImpl* lt,
+    std::shared_ptr<torch::jit::Graph> g,
+    std::map<torch::jit::Value*, c10::IValue>& value_map) {
+  if (!lt->schema_) {
+    auto* v = g->addInput();
+    value_map[v] = IValue(lt->inps_[0].toTensor());
+    return v;
+  }
+  auto fs = torch::jit::parseSchema(lt->schema_);
+  auto s = Symbol::fromQualString(fs.name());
+  auto* node = g->create(s);
+  auto* output = node->output();
+
+  auto i = 0;
+  for (const auto& inp_ : lt->inps_) {
+    torch::jit::Value* v;
+    if (inp_.isTensor()) {
+      auto t = inp_.toTensor();
+      if (c10::tensorTypeIdToBackend(t.type_id()) == Backend::Lazy) {
```

Comment:
Didn't you add a `t.is_lazy()` specifically to not have to type all this out? XD